src/Bicep.Core/TypeSystem/TypeKind.cs

```diff
@@ -37,6 +37,16 @@ public enum TypeKind
         /// <summary>
         /// Union of multiple types.
         /// </summary>
-        Union
+        Union,
+
+        /// <summary>
+        /// String literal type.
+        /// </summary>
+        StringLiteral,
```

Comment:
>StringLiteral [](start = 8, length = 13)

Doesn't this overlap with `Primitive`?

Reply:
The code is also interpreting `Primitive` to mean 'one of the predefined types that can be specified for a parameter/output', which prevented me from using `Primitive`. Conceptually it's a little weird, and there is no code that actually depends on this particular TypeKind (apart from code that depends on it *not* being some other TypeKind).

Reply:
Yeah we probably need to reorganize these type kinds at some point. No issue keeping as-is for now...